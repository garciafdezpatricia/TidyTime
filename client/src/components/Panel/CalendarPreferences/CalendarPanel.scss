@use "../../../../styles/variables" as *;

.calendar-settings {
    padding: 1rem;
    display: flex;
    flex-direction: column;
    width: calc(100% - 2rem);

    p {
        font-size: .9rem;
    }

    .week-start, .event-view {
        padding-top: 1rem;
        font-size: .9rem;
        margin-left: 1.5rem;
    }

    .week-days-buttons {
        display: flex;
        gap: .8rem;
        padding-top: .5rem;

        button, button.selected-day {
            display: flex;
            align-items: center;
            justify-content: center;
            border: none;
            outline: none;
            width: 2rem;
            height: 2rem;
            padding: .8rem;
            &:hover {
                background-color: $light-blue;
            }
        }

        button.selected-day {
            outline: .1rem solid $primary-blue;
        }
    }

    .event-view-buttons {
        display: flex;
        gap: .8rem;
        padding-top: .5rem;

        button, button.selected-view {
            background-color: $background-button;
            display: flex;
            align-items: center;
            justify-content: center;
            border: none;
            outline: none;
            width: fit-content;
            height: 2rem;
            padding: .8rem;
            &:hover {
                background-color: $light-blue;
            }
        }

        button.selected-view {
            outline: .1rem solid $primary-blue;
        }

    }

}