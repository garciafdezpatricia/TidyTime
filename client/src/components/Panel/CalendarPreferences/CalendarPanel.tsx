import { useState } from "react";



export default function CalendarPanel() {

    const [weekDays, setWeekDays] = useState(["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]);

    return (
        <article className="calendar-settings">
            <h3>Calendar preferences</h3>
            <hr></hr>
            <section className="week-start">
                <p>Starting day of week</p>
                <div className="week-days-buttons">
                    {
                        weekDays.map((day, index) => {
                            return (
                                <button key={index}
                                >
                                    {day.toUpperCase().charAt(0)}
                                </button>
                            )
                        })
                    }
                </div>
            </section>
        </article>
    )
}