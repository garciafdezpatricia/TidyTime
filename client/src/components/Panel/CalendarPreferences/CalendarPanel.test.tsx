import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import CalendarPanel from '@/src/components/Panel/CalendarPreferences/CalendarPanel';

describe('CalendarPanel', () => {
    it('renders without crashing', () => {
      const { container } = render(<CalendarPanel />);
      expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<CalendarPanel />);
        expect(asFragment()).toMatchSnapshot();
      });
  });