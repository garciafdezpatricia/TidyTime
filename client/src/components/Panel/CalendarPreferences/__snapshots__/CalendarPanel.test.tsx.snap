// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarPanel matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="calendar-settings"
  >
    <h3>
      preferences.calendar.title
    </h3>
    <hr />
    <p>
      preferences.calendar.desc
    </p>
    <section
      class="week-start"
    >
      <p>
        preferences.calendar.weekDay
      </p>
      <div
        class="week-days-buttons"
      >
        <button
          class="selected-day"
          title="calendar.weekDays.monday.day"
        >
          calendar.weekDays.monday.dim
        </button>
        <button
          class=""
          title="calendar.weekDays.tuesday.day"
        >
          calendar.weekDays.tuesday.dim
        </button>
        <button
          class=""
          title="calendar.weekDays.wednesday.day"
        >
          calendar.weekDays.wednesday.dim
        </button>
        <button
          class=""
          title="calendar.weekDays.thursday.day"
        >
          calendar.weekDays.thursday.dim
        </button>
        <button
          class=""
          title="calendar.weekDays.friday.day"
        >
          calendar.weekDays.friday.dim
        </button>
        <button
          class=""
          title="calendar.weekDays.saturday.day"
        >
          calendar.weekDays.saturday.dim
        </button>
        <button
          class=""
          title="calendar.weekDays.sunday.day"
        >
          calendar.weekDays.sunday.dim
        </button>
      </div>
    </section>
    <section
      class="event-view"
    >
      <p>
        preferences.calendar.calendarView
      </p>
      <div
        class="event-view-buttons"
      >
        <button
          class=""
          title="calendar.messages.month"
        >
          calendar.messages.month
        </button>
        <button
          class=""
          title="calendar.messages.week"
        >
          calendar.messages.week
        </button>
        <button
          class=""
          title="calendar.messages.day"
        >
          calendar.messages.day
        </button>
        <button
          class=""
          title="calendar.messages.agenda"
        >
          calendar.messages.agenda
        </button>
      </div>
    </section>
  </article>
</DocumentFragment>
`;
